src/core/qgsexpressionparser.yy

```diff
@@ -193,22 +195,28 @@ expression:
           {
             // this should not actually happen because already in lexer we check whether an identifier is a known function
             // (if the name is not known the token is parsed as a column)
-            exp_error(parser_ctx, "Function is not known");
+            QgsExpression::ParserError::ParserErrorType errorType = QgsExpression::ParserError::FunctionUnknown;
+            parser_ctx->parserError.errorType = errorType;
+            exp_error(&yyloc, parser_ctx, "Function is not known");
```

Comment:
I don't know if this is the right place in the commit (sorry!) but could these messages be translatable. the `unexpected end` message is bad but it's really worse when you do not speak English (and we have a couple of places where error messages are only shown in English).

Reply:
@DelazJ Yeah they should be.  I will look into that in another PR as I have lots of stuff to change in here at the moment so just doing in bits.